benchmark/BackendBench/eval.py

```diff
@@ -0,0 +1,547 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Evaluation script for KernelAgent using BackendBench infrastructure with a two-phase workflow:
+
+1. Kernel-generation Phase: Generate kernels using KernelAgent → save to generated_kernels/
+2. Evaluation Phase: Run BackendBench main.py with DirectoryBackend to evaluate
+
+"""
+
+import argparse
+import logging
+import os
+import subprocess
+import sys
+from datetime import datetime
+
+import torch
+from BackendBench.suite import OpInfoTestSuite
+from BackendBench.utils import extract_operator_name, op_name_to_folder_name
+from triton_kernel_agent import TritonKernelAgent
+
+
+def generate_kernels(
+    suite_name: str = "torchbench",
+    num_operators: int = None,
+    num_workers: int = 4,
+    max_rounds: int = 10,
+    workflow: str = "base",
+    verbose: bool = False,
+    test_cases: bool = True,
+) -> str:
+    """
+    Phase 1: Generate kernels using TritonKernelAgent.
+
+    Args:
+        suite_name: Test suite name (opinfo/smoke/torchbench)
+        num_operators: Number of operators to generate (None = all)
+        num_workers: Number of parallel workers for KernelAgent
+        max_rounds: Max refinement rounds per worker
+        verbose: Verbose logging
+
+    Returns:
+        Path to generated_kernels directory
+    """
+    logger = logging.getLogger("generate_kernels")
+    logger.info("Phase 1: Generating kernels with TritonKernelAgent")
+
+    # Use standard BackendBench directory structure: generated_kernels/
+    kernels_dir = "generated_kernels"
+    os.makedirs(kernels_dir, exist_ok=True)
+    logger.info(f"Kernels will be saved to: {kernels_dir}/<op_name>/")
+
+    # Initialize TritonKernelAgent with timestamped log directory
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    agent_log_dir = f"agent_logs/run_{timestamp}"
+    os.makedirs(agent_log_dir, exist_ok=True)
+
+    agent = TritonKernelAgent(
+        log_dir=agent_log_dir,
+        num_workers=num_workers,
+        max_rounds=max_rounds,
+    )
+    logger.info(
+        f"TritonKernelAgent initialized (workers={num_workers}, max_rounds={max_rounds})"
+    )
+    logger.info(f"Agent logs: {agent_log_dir}")
+
+    # Create test suite
+    if suite_name == "opinfo":
+        test_suite = OpInfoTestSuite("opinfo", "cuda", torch.bfloat16)
+    elif suite_name == "smoke":
+        from BackendBench.suite import SmokeTestSuite
+
+        test_suite = SmokeTestSuite("smoke", "cuda")
+    elif suite_name == "torchbench":
+        from BackendBench.suite import TorchBenchTestSuite
+
+        test_suite = TorchBenchTestSuite("torchbench", None)
+    else:
+        raise ValueError(f"Unknown suite: {suite_name}")
+
+    # Get operators to generate
+    operators = list(test_suite)
```

Comment:
Might make sense for this to be a static artifact we refresh if backendbench isn't making huge updates